### DISCUSSION TITLE
What IDEs (development environments) is everyone planning on using for this class?

### DISCUSSION THREAD

**Discussion by bst3ck** [USU-ITLS-x265-23-summer/class-repo#21] [Q&A]
As we begin this class, is there a recommended IDE we should use for our assignments? I have used Sublime and Netbeans in the past, but I have started to see more people using Visual Studio code (VSC). I have played around with VSC a little but am unsure how well it is integrated with GitHub. I have not seen any instruction on this in the syllabus or any of the read-me files, so I thought I would open this question up to the full class.

**Comment by metasean** [comment]
**If you're not already comfortable working in an IDE, do _not_ worry about IDEs or Codespaces yet! :wink:**

We won't be needing an IDE until individual project work starts (week 7), but one of the reasons for using GitHub Classroom is that it includes free access to Codespaces, which is basically VS Code in the GitHub cloud.

**_IF_** you are _already_ comfortable working in an IDE, you can access Codespaces using the following screenshot as a guide. 

<img width="800" alt="Screen Shot 2023-05-13 at 1 36 46 PM" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/7552754/911196fa-9ecd-41d7-ac73-00f0fd59db04">

---

**_NOTE_**

You can create multiple codespaces on the same branch!  This can cause a lot of confusion!
When a new Codespace is created, it is automatically given a new name.  Once that exists, you _probably_ want to keep working just in that codespace instance.  For example, in the following screenshot, I created a Codespace that was auto-named "legendary parakeet", so if I want to continue working in that codespace, then I need to click the codespace's name (i.e "legendary parakeet") to open that specific codespace instance.

<img width="433" alt="Screen Shot 2023-05-13 at 1 42 13 PM" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/7552754/b15a2f22-5623-43dc-aca1-d6afcb3c0116">

**Comment by kassyoveson** [comment]
I ended up using Visual Studio Code. I liked it because I was able to view what my project looked like. It was very helpful when I was attempting to do my CSS code. I had a neighbor come and help me upload it to the GitHub repo. I wish there was more instruction on how to connect the 2 programs. That was a very frustrating experience, and I did shed a few tears.
